opm/grid/common/MetisPartition.cpp

```diff
@@ -0,0 +1,214 @@
+        METIS_SetDefaultOptions(options);
+
+        // This is the array of options as described in Section 5.4.
+        // The following options are valid for METIS PartGraphRecursive:
+        // METIS_OPTION_CTYPE, METIS_OPTION_IPTYPE, METIS_OPTION_RTYPE,
+        // METIS_OPTION_NO2HOP, METIS_OPTION_NCUTS, METIS_OPTION_NITER,
+        // METIS_OPTION_SEED, METIS_OPTION_UFACTOR, METIS_OPTION_NUMBERING,
+        // METIS_OPTION_DBGLVL
+        // The following options are valid for METIS PartGraphKway:
+        // METIS_OPTION_OBJTYPE = this is edgecut by default -> minimize the edges that straddle partitions
+        //                        as opposed to vol -> minimize Total communication volume.
+        // METIS_OPTION_CTYPE, METIS_OPTION_IPTYPE,
+        // METIS_OPTION_RTYPE, METIS_OPTION_NO2HOP, METIS_OPTION_NCUTS,
+        // METIS_OPTION_NITER, METIS_OPTION_UFACTOR, METIS_OPTION_MINCONN,
+        // METIS_OPTION_CONTIG, METIS_OPTION_SEED, METIS_OPTION_NUMBERING,
+        // METIS_OPTION_DBGLVL
+
+        //Now, we define all variables that *do depend* on whether there are wells or not
+        idx_t m; // First of all: The number of edges depends on whether there are wells or not 
+
+        std::shared_ptr<CombinedGridWellGraph> gridAndWells;
+
+        if( wells )
+        {
+            gridAndWells.reset(new CombinedGridWellGraph(cpgrid,
+                                                           wells,
+                                                           transmissibilities,
+                                                           true,
+                                                           edgeWeightsMethod));
+            for (int i = 0; i < n;  i++ ) {
+                xadj[i+1] = xadj[i] + Dune::cpgrid::getSpecificEntryOfCpGridWellsNumEdgesList(*gridAndWells, lids, i);
+            }
+        }
+        else
+        {
+            //todo: save this at the right spot, for now, lids[i] = i and gids[i] = i, but thats not always the case, probably
+            for (int i = 0; i < n;  i++ ) {
+                xadj[i+1] = xadj[i] + Dune::cpgrid::getSpecificEntryOfCpGridNumEdgesList(cpgrid, lids[i]);
+            }
+        }
+        m = xadj[n]*0.5;
+
+        idx_t* adjncy = new idx_t[2*m]; //An array that contains the adjacency list of the graph.
+        // The xadj array is of size n + 1 whereas the adjncy array is of size 2m (because for each edge between vertices v and u we actually store both (v, u) and (u, v)).
+        // The adjacency list of vertex i is stored in array adjncy starting at index xadj[i] and ending at (but not
+        // including) index xadj[i + 1] (i.e., adjncy[xadj[i]] through and including adjncy[xadj[i + 1]-1])
+        // So: xadj contains the indices where we start for the respective component
+        
+        idx_t* vsize = new idx_t[2*m]; // The size of the vertices for computing the total communication volume as described in Section 5.7. 
```

Comment:
Size should be equal to the number of vertices. For the entries can be a constant value `vsize[i]==k` for any i as the amount of data per cell that we sent is the same for each cell.

I guess just using NULL for this pointer is OK.